ddsrecorder/test/blackbox/mcap/McapFileCreationTestWithController.cpp

```diff
@@ -0,0 +1,585 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#define MCAP_IMPLEMENTATION
+
+#include <cpp_utils/testing/gtest_aux.hpp>
+#include <cpp_utils/event/FileWatcherHandler.hpp>
+#include <cpp_utils/event/MultipleEventHandler.hpp>
+#include <cpp_utils/event/PeriodicEventHandler.hpp>
+#include <cpp_utils/event/SignalEventHandler.hpp>
+#include <gtest/gtest.h>
+
+#include <cpp_utils/logging/CustomStdLogConsumer.hpp>
+
+#include <ddspipe_core/core/DdsPipe.hpp>
+#include <ddspipe_core/efficiency/payload/FastPayloadPool.hpp>
+
+#include <ddspipe_participants/participant/dynamic_types/DynTypesParticipant.hpp>
+#include <ddspipe_participants/participant/dynamic_types/SchemaParticipant.hpp>
+
+#include <ddsrecorder_participants/mcap/McapHandler.hpp>
+#include <ddsrecorder_participants/mcap/McapHandlerConfiguration.hpp>
+
+#include <ddsrecorder_yaml/YamlReaderConfiguration.hpp>
+#include <ddsrecorder_yaml/yaml_configuration_tags.hpp>
+
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastrtps/types/DynamicDataPtr.h>
+#include <fastrtps/types/DynamicType.h>
+#include <fastrtps/types/DynamicDataFactory.h>
+#include <fastrtps/types/TypeObjectFactory.h>
+
+#include <mcap/reader.hpp>
+
+#include "command_receiver/CommandReceiver.hpp"
+
+#include "types/hello_world/HelloWorldTypeObject.h"
+#include "types/hello_world/HelloWorldPubSubTypes.h"
+
+#include <iostream>
+
+using namespace eprosima::fastdds::dds;
+using namespace eprosima::ddspipe;
+using namespace eprosima::ddsrecorder;
+
+using CommandCode = eprosima::ddsrecorder::receiver::CommandCode;
+using CommandCodeBuilder = eprosima::ddsrecorder::receiver::CommandCodeBuilder;
+using McapHandlerState = eprosima::ddsrecorder::participants::McapHandler::StateCode;
+
+enum class DataTypeKind
+{
+    HELLO_WORLD,
+};
+
+namespace test {
+
+// Publisher
+
+unsigned int domain = 223;
+
+std::string topic = "TypeIntrospectionTopic";
+std::string data_type_name = "HelloWorld";
+
+eprosima::fastdds::dds::DataWriter* writer_;
+eprosima::fastrtps::types::DynamicType_ptr dynamic_type_;
+
+// Configuration
+
+std::vector<const char*> yml_configurations =
+{
+    R"(
+    dds:
+        domain: 223
+    recorder:
+        buffer-size: 3
+        event-window: 3
+    remote-controller:
+        enable: true
+        domain: 222
+    specs:
+        threads: 8
+        max-depth: 100
+        max-pending-samples: 10
+        cleanup-period: 3
+
+    )",
+};
+
+YAML::Node yml;
+
+} // test
+
+std::unique_ptr<core::DdsPipe> create_recorder(
```

Comment:
Maybe move this function to another file, so it is not duplicated. Another option is to merge both test files into one.

Reply:
Same for create_publisher